packages/docusaurus-theme-classic/src/theme/NavbarItem/DocNavbarItem.tsx

```diff
@@ -56,7 +56,7 @@ export default function DocNavbarItem({
       {...props}
       className={clsx(props.className, {
         [activeDocInfimaClassName]:
-          activeDoc && activeDoc.sidebar === doc.sidebar,
+          doc.sidebar && doc.sidebar === activeDoc?.sidebar,
```

Comment:
Let's be closer to original condition and rewrite to something like this:

```suggestion
          activeDoc?.sidebar && activeDoc.sidebar === doc.sidebar,
```